Mathlib/AlgebraicTopology/SimplicialSet/Coskeletal.lean

```diff
@@ -0,0 +1,441 @@
+    let tri₀ : tri' ⟶ pt j :=
+      StructuredArrow.homMk (.op (const [0] _ 0)) (Quiver.Hom.unop_inj (Hom.ext_zero_left ..))
+    let tri₁ : tri' ⟶ pt k₀.castSucc :=
+      StructuredArrow.homMk (.op (const [0] _ 1)) (Quiver.Hom.unop_inj (Hom.ext_zero_left ..))
+    let tri₂ : tri' ⟶ pt k₀.succ :=
+      StructuredArrow.homMk (.op (const [0] _ 2)) (Quiver.Hom.unop_inj (Hom.ext_zero_left ..))
+    have nat := congr_fun (s.π.naturality facemap₁) x
+    simp only [Int.reduceNeg, id_eq, Int.Nat.cast_ofNat_Int, homOfLE_leOfHom,
+      comp_obj, StructuredArrow.proj_obj, op_obj, const_obj_obj, const_obj_map, types_comp_apply,
+      types_id_apply, Functor.comp_map, StructuredArrow.proj_map, op_map] at nat
+    rw [nat]
+    unfold facemap₁
+    simp only [StructuredArrow.homMk_right, Quiver.Hom.unop_op]
+    rw [show mkOfLe _ _ hjk.le = (δ 1) ≫ tri from Hom.ext_one_left .., op_comp, Functor.map_comp]
+    refine congrArg (X.map (δ 1).op) ?_
+    apply StrictSegal.spineInjective
+    unfold StrictSegal.spineEquiv
+    dsimp
+    ext i'
+    · simp only [spine_vertex, ← FunctorToTypes.map_comp_apply, ← op_comp]
+      rw [show [0].const [2] i' ≫ tri = [0].const [n] (([0].const [2] i' ≫ tri).toOrderHom 0) from
+        eq_const_of_zero _]
+      simp only [ran.lift, StrictSegal.spineToSimplex_vertex]
+      match i' with
+      | 0 => exact congr_fun (s.π.naturality tri₀) x
+      | 1 => exact congr_fun (s.π.naturality tri₁) x
+      | 2 => exact congr_fun (s.π.naturality tri₂) x
+    · simp only [spine_arrow, ← FunctorToTypes.map_comp_apply, ← op_comp]
+      match i' with
+      | 0 =>
+        rw [show mkOfSucc 0 ≫ tri = mkOfLe j k₀.castSucc hjk₀ from Hom.ext_one_left ..]
+        have eq' := congr_fun (s.π.naturality facemap₂) x
+        unfold facemap₂ at eq'
+        simp only [Int.reduceNeg, homOfLE_leOfHom, comp_obj, StructuredArrow.proj_obj, op_obj,
+          const_obj_obj, len_mk, Fin.isValue, const_obj_map, types_comp_apply, types_id_apply,
+          Functor.comp_map, StructuredArrow.proj_map, StructuredArrow.homMk_right, op_map,
+          Quiver.Hom.unop_op] at eq'
+        rw [show (δ 2 : [1] ⟶ [2]) = mkOfSucc 0 from Hom.ext_one_left ..] at eq'
+        simp [Truncated.inclusion] at eq'
+        rw [← eq']
+        exact ih k₀.castSucc hjk₀.hom (Nat.succ_inj.1 hi)
+      | 1 =>
+        rw [show mkOfSucc 1 ≫ tri = mkOfSucc k₀ from Hom.ext_one_left ..]
+        simp only [StrictSegal.spineToSimplex_arrow]
+        have := congr_fun (s.π.naturality facemap₀) x
+        dsimp [facemap₀] at this
+        rw [this, show (δ 0 : [1] ⟶ [2]) = mkOfSucc 1 from Hom.ext_one_left ..]
+        rfl
+
+namespace StrictSegal
+variable (X : SSet.{u}) [StrictSegal X]
+
+/-- A strict Segal simplicial set is 2-coskeletal. -/
+noncomputable def IsPointwiseRightKanExtensionAt (n : ℕ) :
+    (rightExtensionInclusion X 2).IsPointwiseRightKanExtensionAt ⟨[n]⟩ := by
+  show IsLimit _
```

Comment:
When defining data, it is not recommended to apply too many tactics, especially `simp`. The structure of the code should be like this:
```suggestion
    (rightExtensionInclusion X 2).IsPointwiseRightKanExtensionAt ⟨[n]⟩ where
  lift s x := ran.lift (X := X) s x
  fac := sorry
  uniq := sorry
```

Reply:
Done (with your branch).